pkg/sdn/plugin/proxy.go

```diff
@@ -87,29 +93,73 @@ func (proxy *OsdnProxy) watchEgressNetworkPolicies() {
 	})
 }
 
-func (proxy *OsdnProxy) updateNetworkPolicy(policy osapi.EgressNetworkPolicy) {
-	firewall := make([]proxyFirewallItem, len(policy.Spec.Egress))
+func (proxy *OsdnProxy) updateEgressNetworkPolicy(policy osapi.EgressNetworkPolicy) {
+	ns := policy.Namespace
+	firewall := make([]firewallItem, len(policy.Spec.Egress))
 	for i, rule := range policy.Spec.Egress {
 		_, cidr, err := net.ParseCIDR(rule.To.CIDRSelector)
 		if err != nil {
 			// should have been caught by validation
-			glog.Errorf("Illegal CIDR value %q in EgressNetworkPolicy rule", rule.To.CIDRSelector)
+			glog.Errorf("Illegal CIDR value %q in EgressNetworkPolicy rule for policy: %v", rule.To.CIDRSelector, policy.UID)
 			return
 		}
-		firewall[i] = proxyFirewallItem{rule.Type, cidr}
+		firewall[i] = firewallItem{rule.Type, cidr}
 	}
 
+	// Add/Update/Delete firewall rules for the namespace
 	if len(firewall) > 0 {
-		proxy.firewall[policy.Namespace] = firewall
+		updated := false
+		if ref, ok := proxy.firewall[ns]; ok {
+			for uid := range ref.namespaceFirewalls {
+				if uid == policy.UID {
```

Comment:
don't need a loop now that you're using a map; just `if _, ok := ref.namespaceFirewalls[policy.UID]; ok {`

Although you can get rid of `updated` in that case too. Just check if `proxy.firewall[ns]` exists, and create it if not, and then set `proxy.firewall[ns].namespaceFirewalls[policy.UID]`